src/components/account/edit/Immatriculation.vue

```diff
@@ -0,0 +1,162 @@
+                required: true,
+            },
+        },
+        setup(props: any) {
+            let refreshKey = ref(false);
+            let busy = ref(true);
+            let fieldsOfStudy = Object.values(FieldOfStudy).filter((e) => e != FieldOfStudy.NONE);
+            let semesterType = ref("");
+            let year = ref("");
+            let selectedFieldsOfStudy = ref([] as FieldOfStudy[]);
+
+            let currentYear = new Date().getFullYear();
+            let selectableYears = computed(() => {
+                let array: String[] = [];
+                for (let index = currentYear; index >= currentYear - 10; index--) {
+                    if (semesterType.value == "WS") {
+                        array.push(
+                            index.toString() + "/" + (index + 1).toString().substring(index.toString().length - 2, index.toString().length)
+                        );
+                    } else {
+                        array.push(index.toString());
+                    }
+                }
+                return array;
+            });
+
+            function updateSelectedFieldsOfStudy(value: any) {
+                selectedFieldsOfStudy.value = value.value;
+            }
+
+            let selectedSemester = computed(() => {
+                return semesterType.value + year.value;
+            });
+
+            let validSelection = computed(() => {
+                return (
+                    semesterType.value != "" &&
+                    year.value != "" &&
+                    selectedFieldsOfStudy.value.length != 0 &&
+                    selectedFieldsOfStudy.value[0] != ""
+                );
+            });
+
+            function resetYear() {
+                year.value = "";
+            }
+
+            async function updateImmatriculation() {
+                busy.value = true;
+                let error = false;
+                let successfullUpdates: number[] = [];
+                const matriculationManagement: MatriculationManagement = new MatriculationManagement();
+
+                //TODO Replace the for-loop with a single call containing a list of FoSs  as soon as provided by backend
+                for (let i = 0; i < selectedFieldsOfStudy.value.length; i++) {
+                    const response = await matriculationManagement.updateMatriculationData(
+                        props.username,
+                        selectedFieldsOfStudy.value[i],
+                        selectedSemester.value
+                    );
+                    const responseHandler = new GenericResponseHandler();
+                    const result = responseHandler.handleReponse(response);
+                    if (response.statusCode != 200) {
+                        error = true;
+                        console.log(
+                            "Error: Updating Entry '" + selectedSemester.value + " : " + selectedFieldsOfStudy.value[i] + "' Failed!"
+                        );
```

Comment:
see other comment

Reply:
This section will be changed at that point when the backend provides an endpoint for submitting a list of FoSs per semester. ATM this is needed to detect failing submits